rotkehlchen/api/rest.py

```diff
@@ -3474,9 +3474,16 @@ def get_history_events(
                 action_type=ActionType.HISTORY_EVENT,
             )
 
+        valid_entry_combinations = DBAccountingRules(self.rotkehlchen.data.db).get_valid_entry_combinations(events_result)  # noqa: E501
+
```

Comment:
for this I would process the events here since you know wether they are a list or a list of tuples and do something like

```python
if group_by_events_ids is False:
    get_valid_entry_combinations(events_result)
else:
    get_valid_entry_combinations([event for _, event in events_result])
```

you can call the method in the if statement below to not check again. The reason is that this way you simplify the logic in `get_valid_entry_combinations` and you adjust the data to the function instead of the function to the data format